packages/notification-center/src/components/notification-center/NotificationCenter.tsx

```diff
@@ -25,33 +25,51 @@ export interface INotificationCenterProps {
   onTabClick?: (tab: ITab) => void;
 }
 
-export function NotificationCenter(props: INotificationCenterProps) {
-  const queryClient = new QueryClient();
+export function NotificationCenter({
+  onUnseenCountChanged,
+  onUrlChange,
+  onNotificationClick,
+  onActionClick,
+  header,
+  footer,
+  emptyState,
+  listItem,
+  actionsResultBlock,
+  tabs,
+  showUserPreferences,
+  onTabClick,
+  colorScheme,
+  theme,
+}: INotificationCenterProps) {
   const { applicationIdentifier } = useNovuContext();
+  const { unseenCount } = useNotifications();
+
+  useEffect(() => {
+    if (onUnseenCountChanged) {
+      onUnseenCountChanged(unseenCount);
+    }
+  }, [unseenCount, (window as any).parentIFrame]);
```

Comment:
This code was moved from the `Header.tsx`, and ideally, it should be in the dependency array. I don't want to add that function directly to the deps array because right now it might cause an infinite re-renders loop if someone does something like this `onUnseenCountChanged={(count) => { ...code }}`.
But I'll try to improve it with a ref call instead.